tgui/packages/tgui/layouts/Window.js

```diff
@@ -24,9 +24,15 @@ const DEFAULT_SIZE = [400, 600];
 export class Window extends Component {
   componentDidMount() {
     const { suspended } = useBackend(this.context);
+    const { canClose } = this.props;
     if (suspended) {
       return;
     }
+    if (!canClose) {
+      Byond.winset(window.__windowId__, {
+        'can-close': false,
+      });
+    }
```

Comment:
This is bad, because this window will never restore to `can-close: true` once it's set to `false`.

This would be much better, even though it is slightly more expensive to always call it, but it is possible to avoid the call by storing the currently set value as a variable, and only call it when it changes.

```jsx
Byond.winset(window.__windowId__, {
  'can-close': Boolean(canClose),
});
```